### ISSUE TITLE
HomeKit Switch Identification

### ISSUE THREAD

**Issue by akagarwal** [dustindclark/homebridge-envisalink#287]
Three panic switches and two alarm switches get added in HomeKit regardless of true or false setup in HB config which only either enables or disabled the functionality. It is probably best to enable everything in order not to create a false impression that they are operational. The issue is that there are no names and no way to identify them. For alarm switches, however, I was able to identify the chime switch but the 2nd switch took some reading, guessing and playing to figure out that it was the open zone bypass switch. So, I am okay there. For panic switches however, they just show up as Switch, Switch 2 and Switch 3. Can you please help with how to identify these? I definitely do not want to turn these on to figure it out. Also, do these simply work by turning them on or will there be a confirmation or code to entered etc.?

**Comment by brownjd** [comment]
Agreed. If you disable the functionality, there is no reason to have the buttons show up. I don't even see the panic switches.

**Comment by dustindclark** [comment]
This is a HomeKit bug, but I've finally found a hack in iOS 17 that allows you to identify them. Tap on your alarm in the Home app, then tap on the Settings Gear > Accessories. Tap on each switch (not the switch itself, but to the right of that). Then tap the Settings gear on that switch. Clear the name here, and the underlying (real) name will appear grayed out in the background. You should be able to name it something sensible here as well.

**Comment by brownjd** [comment]
But, can we just disable the buttons if we don't use them?

**Comment by dustindclark** [comment]
Yes. See readme.

**Comment by brownjd** [comment]
I did. I see these. How do you disable zone bypass?

"ambulancePanic": {
          "enabled": true,
          "name": "Ambulance Panic"
      },
      "firePanic": {
          "enabled": true,
          "name": "Fire Panic"
      },
      "policePanic": {
          "enabled": true,
          "name": "Police Panic"
      },
"partitions": [
        {
          "name": "Alarm",
          "enableChimeSwitch": true,
          "pin": "1243"
        }
      ],

**Comment by dustindclark** [comment]
Zone bypass is the only switch that cannot be disabled. The reasoning here is that you cannot set your alarm remotely or via automation if it is obstructed (door or window open). The bypass switch enables that.

Open to a compelling argument and/or a PR for allowing it to be disabled.

**Comment by brownjd** [comment]
So how does it work? I seem to recall that if you want to bypass a zone from the panel when arming, you have to specify the zone. With the UI simply exposing a switch, how does it know what zone to bypass? Is it just whatever zone is open at the time?

**Comment by dustindclark** [comment]
That's correct. If the switch is on, it bypasses any obstructed zone when arming.

**Comment by brownjd** [comment]
Okay, I agree with your reasoning. You need to be able to remotely set the alarm even if a zone is obstructed.
